### PR TITLE
Add tags datasets for LB radio

### PR DESCRIPTION
**Pull request by amCap1712**
Adds the LB radio tags dataset.

### CODE REVIEW

File: listenbrainz/db/tags.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            curs.execute(query)
+            query = SQL("""DROP TABLE {old_table}""").format(old_table=Identifier("tags", "tags_old"))
+            curs.execute(query)
+        conn.commit()
+    finally:
+        conn.close()
+
+
+def insert(source, recordings):
+    values = []
```
Comment (mayhem) [inline listenbrainz/db/tags.py:73]: docstrings

File: listenbrainz/webserver/views/api.py [hunk 2]
```diff
@@ -665,6 +665,64 @@ def get_service_details(user_name):
...
+    if len(tag) == 1 and condition is None:
+        condition = "OR"
+
+    if condition is None:
+        raise APIBadRequest("multiple tags are specified but the condition param is missing")
+    condition = condition.upper()
+    if condition != "AND" and condition != "OR":
+        raise APIBadRequest("condition param should be either 'AND' or 'OR'")
+
+    try:
```
Comment (mayhem) [inline listenbrainz/webserver/views/api.py:689]: I think for an API we should use percent values from 0 - 100 -- it little more intuitive for people, I think.

**Review by mayhem — CHANGES_REQUESTED** [review]
Ok looks pretty good so far. Some naming changes and some missing docstrings, but overall great work!

File: listenbrainz/webserver/views/api.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz/webserver/views/api.py:689]: done

File: listenbrainz/db/tags.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/db/tags.py:74]: done

File: listenbrainz/spark/handlers.py [hunk 3]
```diff
@@ -465,3 +465,15 @@ def handle_troi_playlists(message):
 
 def handle_troi_playlists_end(message):
     batch_process_playlists_end(message["slug"])
+
+
+def handle_tags_dataset_start(message):
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply listenbrainz/spark/handlers.py]: done

File: listenbrainz/db/timescale.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def init_db_connection(connect_str):
 
     while True:
         try:
-            engine = create_engine(connect_str, poolclass=NullPool)
+            engine = create_engine(connect_str, echo=True, poolclass=NullPool)
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply listenbrainz/db/timescale.py]: removed

File: admin/timescale/create_tables.sql [hunk 5]
```diff
@@ -219,4 +219,12 @@ CREATE TABLE similarity.artist_credit_mbids (
     score INT NOT NULL
 );
 
+CREATE TABLE tags.tags (
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply admin/timescale/create_tables.sql]: done

File: listenbrainz/webserver/views/api.py [hunk 6]
```diff
@@ -665,6 +665,64 @@ def get_service_details(user_name):
     return jsonify({'user_name': user_name, 'services': services})
 
 
+@api_bp.route("/tags/", methods=['GET', 'OPTIONS'])
+@crossdomain
+@ratelimit()
+def get_tags_dataset():
+    """ Get the recordings with the specified tag. """
```
Comment (amCap1712) [inline listenbrainz/webserver/views/api.py:672]: We need to finalize the url for this endpoint before merging. Also add some more docs here methinks.

Comment (mayhem) [reply listenbrainz/webserver/views/api.py:672]: Agreed.

**Review by mayhem — APPROVED** [review]
Great, lets change the endpoint name, add the improved docs and merge!

**Pull request merged**